### PR TITLE
[195] Implement rimble UI

### PR DESCRIPTION
**Pull request by iooeduard**
image

### CODE REVIEW

File: packages/react-app/src/services/ethers-contract.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { ethers } from 'ethers';
+import Abi from '../abi/freaks-abi.json';
+
+export async function getContract() {
+  const deployedAddress = '0x5FbDB2315678afecb367f032d93F642f64180aa3';
```
Comment (dasilvaluis) [inline packages/react-app/src/services/ethers-contract.js]: Can we start extracting these values?
Which are user/session dependent and which are environment?

The environment should be loaded using a .env file, as shown here:
https://tacomanator.medium.com/environments-with-create-react-app-7b645312c09d

User and session values should be received through function arguments.

File: packages/react-app/src/helpers/contract-parser.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export function parseFreaks({ _name, _scor }) {
```
Comment (dasilvaluis) [inline packages/react-app/src/helpers/contract-parser.js]: Consider renaming the `_scor` on the smart contract, it seems to be a typo

File: packages/react-app/src/components/FreaksList/freaks-list.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  function renderFreak(freak, index) {
+    const { name, share } = freak;
+    const sharePercentage = toPercentage(share);
+
+    return (
+      <tr key={ index }>
+        <td>{ name }</td>
+        <td>
+          { sharePercentage }
```
Comment (dasilvaluis) [inline packages/react-app/src/components/FreaksList/freaks-list.jsx]: Maybe it is easier as

```suggestion
          { `${ sharePercentage } %` }
```

**Review by dasilvaluis — CHANGES_REQUESTED** [review]

**Review by dasilvaluis — APPROVED** [review]

**Review by dasilvaluis — CHANGES_REQUESTED** [review]
Re-requesting changes

File: packages/react-app/src/services/ethers-contract.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { ethers } from 'ethers';
+import Abi from '../abi/freaks-abi.json';
+import Web3Service from './web3-service';
+
+export async function getContract() {
+  const deployedAddress = process.env.REACT_APP_CONTRACT_ADDRESS;
+  const localhost = 'http://localhost:8545';
```
Comment (dasilvaluis) [inline packages/react-app/src/services/ethers-contract.js]:
```suggestion
  const providerAddress = 'http://localhost:8545';
```

File: packages/react-app/src/services/ethers-contract.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { ethers } from 'ethers';
+import Abi from '../abi/freaks-abi.json';
+import Web3Service from './web3-service';
+
+export async function getContract() {
+  const deployedAddress = process.env.REACT_APP_CONTRACT_ADDRESS;
```
Comment (dasilvaluis) [inline packages/react-app/src/services/ethers-contract.js]: packages/react-app/src/config.js

const CONTRACT_ADDRESS = process.env.REACT_APP_CONTRACT_ADDRESS;

export default {
  CONTRACT_ADDRESS
};

File: packages/react-app/src/hooks/useEthersContract.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { useEffect, useState } from 'react';
+import { getContract } from '../services/ethers-contract';
+
+export default function useEthersContract() {
+  const [ contract, setContract ] = useState(null);
+
+  function loadContract() {
+    getContract().then(setContract);
```
Comment (dasilvaluis) [inline packages/react-app/src/hooks/useEthersContract.js]: async function ...() {
const contract = await getContract()

setContract(contract);
}

**Review by dasilvaluis — CHANGES_REQUESTED** [review]

**Review by dasilvaluis — APPROVED** [review]

**Comment by iooeduard** [comment]
[Implemet rimble-ui](https://trello.com/c/KHkWhy6i/195-implemet-rimble-ui)

**Pull request merged**